source/slang/hlsl.meta.slang

```diff
@@ -231,28 +295,45 @@ struct ByteAddressBuffer
 
     [__readNone]
     [ForceInline]
-    [require(cpp_cuda_glsl_hlsl_metal_spirv, byteaddressbuffer)]
-    uint3 Load3Aligned(int location)
+    [require(hlsl, byteaddressbuffer)]
+    uint3 Load3(int location, out uint status)
     {
         __target_switch
         {
         case hlsl: __intrinsic_asm ".Load3";
-        default:
-            return __byteAddressBufferLoad<uint3>(this, location, __naturalStrideOf<uint3>());
         }
     }
 
+    /// Load three 32-bit unsigned integers from the buffer at the specified location with alignment
+    /// of `uint3`, which is 4.
+    ///@param location The input address in bytes which must be a multiple of alignment of 4.
+    ///@return `uint3` Three 32-bit unsigned integer value loaded from the buffer.
     [__readNone]
     [ForceInline]
-    [require(hlsl, byteaddressbuffer)]
-    uint3 Load3(int location, out uint status)
+    [require(cpp_cuda_glsl_hlsl_metal_spirv, byteaddressbuffer)]
+    uint3 Load3Aligned(int location)
     {
         __target_switch
         {
         case hlsl: __intrinsic_asm ".Load3";
+        default:
+            return __byteAddressBufferLoad<uint3>(this, location, __naturalStrideOf<uint3>());
         }
     }
 
+    /// Load four 32-bit unsigned integers from the buffer at the specified location.
```

Comment:
this should be 16.

Reply:
uint4 is four 16bits integer?

Reply:
it is four 32 bit integers, and LoadAlign will align to the vector boundary, which is 16 bytes.